homeassistant/components/roborock/image.py

```diff
@@ -64,25 +65,33 @@ def __init__(
         map_flag: int,
         starting_map: bytes,
         map_name: str,
+        roborock_storage: RoborockStorage,
+        create_map: bool,
```

Comment:
Now starting map can either be map bytes or image bytes, and it seems like it would there should be a way to pick one to keep the abstraction here simpler. Can `RoborockMap` just deal with one or the other? Setting a more cirps boundary between the storage class, the coordinator, and the entity will avoid all parts of the system needing to know everything about how the caching works.  (There are also multiple levels of caching happening here now so you can consider how to simplify that in the future as well, putting all the cache in a single layer)

Reply:
Simplified RoborockMap just taking in one type of input. We try to parse beforehand instead of relying on the map object to do it.